src/renderer_gl.cpp

```diff
@@ -1075,6 +1075,189 @@ namespace bgfx { namespace gl
 	{
 	}
 
+	static GLuint currentProgram = 0;
+	static void GlUseProgram(GLuint program)
+	{
+		currentProgram = program;
+		glUseProgram(program);
+	}
+
+	template<typename T>
+	class TinyHashMap
```

Comment:
There are already two different style HashMap implementations in bx or tinystl, pick one that's appropriate.

Reply:
Hmm, I thought to pick the one that performs better. Located `tinystl::unordered_map` but unable to find the bx one - how is that one named?

Reply:
`bx::HandleHashMap*` it doesn't deal with backing storage just deals with indices.

Also if you want to limit storage, see `StateCacheLru`: https://github.com/bkaradzic/bgfx/blob/f07a0f51792db2351f6a38e1625a704be159a6ee/src/renderer.h#L294

Reply:
Thanks. Perhaps HandleHashMap might not be as good fit compared to tinystl::unordered_map, as that would need a second vector to store the actual data.

`StateCacheLru` seems to back to `std::unordered_map`, so that would incur the +7.9% slowdown at least. No indication that runtime memory usage would be a concern here.